File: Yoo.C/뻘겅책/04_21.py
import sys
input = sys.stdin.readline
res=0

def merge_sort(s,e):
    global res
    if e-s<1: return
    m= int(s+(e-s)/2)
    merge_sort(s,m)
    merge_sort(m+1,e)
    
    for i in range(s,e+1):
        tmp[i]= A[i]
        k=s
        idx1= s
        idx2 = m+1
        
    while idx1<= m and idx2<= e:
        if tmp[idx1]> tmp[idx2]:
            A[k]= tmp[idx2]
            res= res+idx2-k
            k+=1
            idx2+=1
        else:
            A[k]= tmp[idx1]
            k+=1
            idx+=1
            
    while idx1<= m:
        A[k]= tmp[idx1]
        k+=1
        idx2+=1

n= int(input())
A=list(map(int, input().split()))

A.insert(0,0)
tmp=[0]* int(n+1)
merge_sort(1,n)
print(res)

File: Yoo.C/뻘겅책/04_2_1.cpp
bool change = false;
for(int i=1; i<=n+1; i++){
    change = false;
    for(int j=1; j<=n-i; j++){
        if(a[j]>a[j+1]){
            change = true;
            int tmp = a[j];
            a[j]=a[j+1];
            a[j+1]= tmp;
        }
    }
    if (change==false){
        cout<<i<<'\n';
        break;
    }

}